responsive-css.md

```diff
@@ -0,0 +1,123 @@
+# Styling your website to be responsive
+
+## Viewport
+
+The viewport is the part of the webpage that's currently visible to the user. Viewports vary between different devices,
+as the screen size between devices also varies. For example, the viewport of a desktop monitor will be significantly larger than the viewport of
+a mobile device.
+
+The viewport can be controlled by setting a `<meta>` tag in each web page you'd like to make responsive. The following is an example of how to use this tag:
+
+```html
+<meta name="viewport" content="width=device-width, initial-scale=1.0" />
+```
+
+`width=device-width` sets the width of the viewport to match the width of the device. This is useful for devices (such as mobile devices) that have less width compared to the height of the device.
+
+`initial-scale` sets the zoom factor of the viewport to the value of the scale provided. We set this to `1.0` to set it as the default zoom for the viewport when the webpage is first loaded.
+
+## Mobile-First Design
+
+Mobile-first design refers to the practice of designing your website for mobile devices before designing your website for desktop devices.
```

Comment:
isn't another reason for doing this so that mobile devices load faster?

Reply:
I guess an argument could be made that a mobile-first design would improve perf by restricting the complexity of your website based on the smaller resolution of mobile devices. More info about it [here](https://sympli.io/blog/2016/07/19/3-vital-ways-a-mobile-first-approach-can-improve-web-performance/).